src/libcore/iter/range.rs

```diff
@@ -36,6 +35,24 @@ pub trait Step: PartialOrd + Sized {
     /// Returns `None` if it is not possible to calculate `steps_between`
     /// without overflow.
     fn steps_between(start: &Self, end: &Self, by: &Self) -> Option<usize>;
+
```

Comment:
In a way, I feel good about this change... feels a bit more clearly purpose-built now.